### PR TITLE
feat: add isPrefetch property in query action

### PR DESCRIPTION
**Pull request by juliengbt**
I am not so sure where and how to write test for it though.

related to #4149 

@EskiMojo14

### CODE REVIEW

**Comment by EskiMojo14** [comment]
my concern with adding a public option is that it allows passing it incorrectly when calling initiate yourself

my thinking would be something a little sneakier, like:
```diff
diff --git a/packages/toolkit/src/query/core/buildInitiate.ts b/packages/toolkit/src/query/core/buildInitiate.ts
index 384d10c5..0d06d8c7 100644
--- a/packages/toolkit/src/query/core/buildInitiate.ts
+++ b/packages/toolkit/src/query/core/buildInitiate.ts
@@ -301,6 +301,7 @@ You must add the middleware for RTK-Query to function correctly!`,
           forceRefetch,
           subscriptionOptions,
           [forceQueryFnSymbol]: forceQueryFn,
+          ...rest
         } = {},
       ) =>
       (dispatch, getState) => {
@@ -319,6 +320,7 @@ You must add the middleware for RTK-Query to function correctly!`,
           originalArgs: arg,
           queryCacheKey,
           [forceQueryFnSymbol]: forceQueryFn,
+          ...rest,
         })
         const selector = (
           api.endpoints[endpointName] as ApiEndpointQuery<any, any>
diff --git a/packages/toolkit/src/query/core/buildThunks.ts b/packages/toolkit/src/query/core/buildThunks.ts
index 2fe616bb..61acf414 100644
--- a/packages/toolkit/src/query/core/buildThunks.ts
+++ b/packages/toolkit/src/query/core/buildThunks.ts
@@ -609,11 +609,12 @@ In the case of an unhandled error, no tags will be "provided" or "invalidated".`
       const force = hasTheForce(options) && options.force
       const maxAge = hasMaxAge(options) && options.ifOlderThan
 
-      const queryAction = (force: boolean = true) =>
-        (api.endpoints[endpointName] as ApiEndpointQuery<any, any>).initiate(
-          arg,
-          { forceRefetch: force },
-        )
+      const queryAction = (forceRefetch = true) => {
+        const options = { forceRefetch, isPrefetch: true }
+        return (
+          api.endpoints[endpointName] as ApiEndpointQuery<any, any>
+        ).initiate(arg, options)
+      }
       const latestStateValue = (
         api.endpoints[endpointName] as ApiEndpointQuery<any, any>
       ).select(arg)(getState())
```

**Comment by juliengbt** [comment]
> my concern with adding a public option is that it allows passing it incorrectly when calling initiate yourself
> 

@EskiMojo14  Moving `isPrefetch` to `StartQueryActionCreatorOptions` isn't going to make private though.
Or were you thinking of forcing the the type of `options` to be `StartQueryActionCreatorOptions` ? (So `isPrefetch` doesn't show up as an option)

```diff
diff --git a/packages/toolkit/src/query/core/buildThunks.ts b/packages/toolkit/src/query/core/buildThunks.ts
index f07e6700..ac280ada 100644
--- a/packages/toolkit/src/query/core/buildThunks.ts
+++ b/packages/toolkit/src/query/core/buildThunks.ts
@@ -613,7 +613,7 @@ In the case of an unhandled error, no tags will be "provided" or "invalidated".`
       const queryAction = (force: boolean = true) =>
         (api.endpoints[endpointName] as ApiEndpointQuery<any, any>).initiate(
           arg,
-          { forceRefetch: force, isPrefetch: true }
+          { forceRefetch: force, isPrefetch: true } as StartQueryActionCreatorOptions
         )
       const latestStateValue = (
         api.endpoints[endpointName] as ApiEndpointQuery<any, any>
```

**Comment by EskiMojo14** [comment]
no cast necessary - if you see in my diff, saving it to a separate variable first means that structural typing kicks in and doesn't care about the extra key (and the ...rest just think it's an empty object)

though in hindsight a cast would probably be a tiny bit more size efficient codewise, so do feel free to do that

**Comment by juliengbt** [comment]
@EskiMojo14 Yes alright! 
Btw I'm sure you're aware of that, just to make sure. Spreading the rest of the options will also allow the user have their own keys in the `meta.arg` object of the start query action, right ? (when calling `initiate`)

**Comment by EskiMojo14** [comment]
yeah - i think that's okay though

File: packages/toolkit/src/query/core/buildInitiate.ts [hunk 1]
```diff
@@ -319,6 +320,7 @@ You must add the middleware for RTK-Query to function correctly!`,
           originalArgs: arg,
           queryCacheKey,
           [forceQueryFnSymbol]: forceQueryFn,
+          ...rest
```
Comment (EskiMojo14) [inline packages/toolkit/src/query/core/buildInitiate.ts:323]: though it's probably worth moving this spread *before* the other parameters, i think

Comment (juliengbt) [reply packages/toolkit/src/query/core/buildInitiate.ts]: Sure

**Comment by EskiMojo14** [comment]
thanks!

**Pull request merged**